### ISSUE TITLE
calculate IoU based on y_true and y_pred?

### ISSUE THREAD

**Issue by graftedlife** [matterport/Mask_RCNN#995]
Right now the loss function related to mask contours (mrcnn_mask_loss_graph) seems to be based on binary crossentropy between the ground truth masks and the predicted masks. Would it improve the final IoUs if we incorporate IoU metrics into the loss based on y_true/y_pred in mrcnn_mask_loss_graph? If so, how?

**Comment by keineahnung2345** [comment]
I've found interesting discussion here: [https://stackoverflow.com/questions/40475246/why-does-one-not-use-iou-for-training](url).
